app/views/branches/_filter.erb

```diff
@@ -0,0 +1,15 @@
+<%= form_tag filter_path(@branch.id), method: :get, remote: true, class: 'filter' do %>
+  <div class="form-group w-75">
+    <label>Filter by day</label>
+    <%= date_field_tag :day, class:"form-date"%>
+  </div>
+  <div class="form-group w-75 my-4">
+    <label>Filter by role</label>
+    <%= select_tag :role_id, options_from_collection_for_select(Role.all, "id", "description"), prompt: "Select a role", class:"form-select"%>
```

Comment:
It is not recommended that you leave model information in the view, I recommend that you call this information from a helper.

```suggestion
    <%= select_tag :role_id, options_from_collection_for_select(get_role, "id", "description"), prompt: "Select a role", class:"form-select"%>
```

In the helper:

```
# app/helpers/branch_helper.rb

module BranchHelper
  def get_role
    Role.all
  end
end
`

Reply:
Thanks, I used it in my code .